agents/meinberg_m1000/meinberg_m1000_agent.py

```diff
@@ -0,0 +1,201 @@
+                            {"oid": ObjectType(ObjectIdentity('MBG-SNMP-LTNG-MIB', 'mbgLtNgPtpPortState', 1)), "interval": 3, "lastGet": None}]
+
+    @inlineCallbacks
+    def start_record(self, session, params=None):
+        """start_record(params=None)
+
+        OCS Process for fetching values from the M1000 via SNMP.
+
+        """
+        if params is None:
+            params = {}
+
+        self.is_streaming = True
+
+        # Determine unique interval values
+        interval_groups = list({x['interval'] for x in self.mib_timings})
+
+        while self.is_streaming:
+            # Loop through interval groups and issue get commands for each
+            # interval group, publishing to a block with the interval in the
+            # name.
+            for interval in interval_groups:
+                # Create list of OIDs to GET based on last time we checked them
+                get_list = []
+                for mib in self.mib_timings:
+                    # Select only OIDs of the same interval
+                    if mib['interval'] != interval:
+                        continue
+
+                    if mib["lastGet"] is None:
+                        get_list.append(mib["oid"])
+                    elif time.time() - mib["lastGet"] > mib["interval"]:
+                        get_list.append(mib["oid"])
+
+                if get_list:
+                    # Issue SNMP GET command
+                    result = yield self.snmp.get(get_list)
+                    read_time = time.time()
+
+                if result:
+                    message = {
+                        'block_name': f'm1000_{interval}',
+                        'timestamp': read_time,
+                        'data': {}
+                    }
+
+                    for item in result:
+                        try:
+                            # OID from SNMP GET
+                            oid = item[0].prettyPrint()
+                            # Makes something like 'MBG-SNMP-LTNG-MIB::mbgLtNgRefclockState.1'
+                            # look like 'mbgLtNgRefclockState_1'
+                            valid_field = oid.split("::")[1].replace('.', '_')
+
+                            self.log.debug("{o} {value}",
+                                           o=oid,
+                                           value=int(item[1]))
+                            message['data'][valid_field] = int(item[1])
+                        except ValueError:
+                            self.log.warn('{oid} is of type {_type}, not int',
+                                          oid=item[1], _type=type(item[1]))
+
+                    # Update lastGet time
+                    for mib in self.mib_timings:
+                        if mib['oid'] in get_list:
+                            mib['lastGet'] = read_time
+
+                    self.log.debug("{msg}", msg=message)
+                    session.app.publish_to_feed('m1000', message)
```

Comment:
Is having the string saved needed? Now with it in session.data the latest string description is visible to clients. Grafana can be configured to display the corresponding string, but through a value mapping setup in a dashboard, rather than storing the strings themselves in InfluxDB. What other applications did you have in mind?